packit_service/worker/jobs.py

```diff
@@ -206,8 +213,16 @@ def process_jobs(self, event: Event) -> Dict[str, HandlerResults]:
                     config=self.config, job_config=job_config, event=event
                 )
                 if handler.pre_check():
-                    handlers_results[job_config.type.value] = handler.run_n_clean()
-            # don't break here, other handlers may react to the same event
+                    event.package_config.set_job_overrides(job_config)
+                    # we can have multiple job definitions with the same type
+
+                    current_time = datetime.datetime.now().strftime(DATETIME_FORMAT)
+                    result_key = f"{job_config.type.value}-{current_time}"
+                    handlers_results[result_key] = handler.run_n_clean()
+                    # once the handler is done, we need to reset package_config
+                    # to the former content because not every job may override the same values
+                    event.package_config.__dict__ = deepcopy(pristine_package_config)
```

Comment:
Would be nice to avoid reverting and don't touch the event's config from handlers.

I know it's a completely different way, but what about this:

- Each handler will work only with `JobConfig`.
- JobConfig will have the same attributes as the `Config`. Each value can be set on the job level but will use the global one if not set.

Reply:
You are asking me to rewrite the whole thing from scratch. I understand the proposed implementation is not ideal, but is the simplest & most efficient way I was able to come up with.

Honestly, I don't understand the benefits your solution brings, aside from a ton of refactoring in p-s and adding support for such a thing in packit.

Reply:
> You are asking me to rewrite the whole thing from scratch.

No, it's either too late or outside of the label-friday scope.

> Honestly, I don't understand the benefits your solution brings, aside from a ton of refactoring in p-s and adding support for such a thing in packit.

- :heavy_plus_sign: cleaner work with the config from handlers
  - You have only one object with all needed info. That can help with the task refactor.
  - You can easily see what value comes from what place.
- :heavy_plus_sign: transparent, you can match `package_config` object with the one from the file (unlike this one) anytime.
- :heavy_plus_sign: overwritting possible without the need for dedicated field => better UX
- :heavy_plus_sign: sharing common values across multiple jobs (e.g. `targets`).
- :heavy_minus_sign: a lot of work to make it properly

It's not doable short term => It's much better to have your implementation than nothing.